src/api/annotation/AnnotationAPI.js

```diff
@@ -0,0 +1,315 @@
+    *          has been created, or be rejected if it cannot be saved
+    */
+    async create({name, domainObject, annotationType, tags, contentText, targets}) {
+        if (!Object.keys(this.ANNOTATION_TYPES).includes(annotationType.toUpperCase())) {
+            throw new Error(`Unknown annotation type: ${annotationType}`);
+        }
+
+        if (!Object.keys(targets).length) {
+            throw new Error(`At least one target is required to create an annotation`);
+        }
+
+        const domainObjectKeyString = this.openmct.objects.makeKeyString(domainObject.identifier);
+        const originalPathObjects = await this.openmct.objects.getOriginalPath(domainObjectKeyString);
+        const originalContextPath = this.openmct.objects.getRelativePath(originalPathObjects);
+        const namespace = domainObject.identifier.namespace;
+        const type = 'annotation';
+        const typeDefinition = this.openmct.types.get(type);
+        const definition = typeDefinition.definition;
+
+        const createdObject = {
+            name,
+            type,
+            identifier: {
+                key: uuid(),
+                namespace
+            },
+            tags,
+            annotationType,
+            contentText,
+            originalContextPath
+        };
+
+        if (definition.initialize) {
+            definition.initialize(createdObject);
+        }
+
+        createdObject.targets = targets;
+        createdObject.originalContextPath = originalContextPath;
+
+        const success = await this.openmct.objects.save(createdObject);
+        if (success) {
+            this.emit('annotationCreated', createdObject);
+
+            return createdObject;
+        } else {
+            throw new Error('Failed to create object');
+        }
+    }
+
+    /**
+    * Get annotations for a given domain object
+    *
+    * @method getAnnotationsForTarget
+    * @param {module:openmct.DomainObject} domainObject the domain object to
+    *        create
+    * @returns {Promise} a promise which will resolve when the domain object
+    *          has been created, or be rejected if it cannot be saved
+    */
+    async getAnnotationsForTarget(targetDomainObject) {
+        if (!targetDomainObject) {
+            throw new Error('😢 Asked to get an annotation, but was passed a null domain object');
+        }
+
+        const targetKeyString = this.openmct.objects.makeKeyString(targetDomainObject.identifier);
+        let foundAnnotations = null;
+
+        const searchResults = (await Promise.all(this.openmct.objects.search(targetKeyString, null, this.openmct.objects.SEARCH_TYPES.ANNOTATIONS))).flat();
+        if (searchResults) {
+            foundAnnotations = searchResults;
+        }
+
+        const composition = await this.openmct.composition.get(targetDomainObject).load();
+
+        Promise.all(composition.map(async childObject => {
```

Comment:
Think we're missing an `await` or something here?

Reply:
Realized this function is basically only used by the plot annotation stuff, so we can remove it for this PR.